units/commit_queue.go

```diff
@@ -135,7 +148,8 @@ func (j *commitQueueJob) Run(ctx context.Context) {
 	}
 
 	// check if a patch has already been created for this PR
-	existingPatch, err := patch.FindOneByGithubPRNum(nextItemInt)
+	mergeCommitSHA := pr.GetMergeCommitSHA()
+	existingPatch, err := patch.FindOneByGithubPRNumAndMergeCommitSHA(nextItemInt, mergeCommitSHA)
```

Comment:
fix to only stop if we've made a patch for this exact merge commit (and keep going even if another patch matches the PR number (e.g. PR testing or an earlier test for the commit queue)